🔀 merge updates from source repo

migbro opened the pull request:
# Pull Request Template

## Description

GATK4 workflow has been refactored to more strictly match the Broad WDL, and to mirror the COV-IRT HEC pipeline, with the exception of genotype-calling specific params. It also incorporates knowledge from the COV-IRT pipeline group, namely Robert Meller in which BQSR should be run on the input bam as a whole, and not split by intervals.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Extensive testing in the Cavatica environment, with output comaprisons on various file splitting conditions show that, without extensive benchmarking, the differing result obtained from haplotype caller when splitting input bams for processing versus not (except for haplotype calling itself) does not produce exact results and can't be confirmed to be "good enough." This pipeline is slower that the previous but adheres to Broad run style.

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

migbro committed:
:hammer: added missing param from Broand WDL

migbro committed:
:arrow_up: upgrade gatk version to 4.1.7.0

migbro committed:
:wrench: since part of wf, just expected .bai for now

migbro committed:
:book: updated GATK versions and HC params in readme. Added rough cost estimates

migbro committed:
:hammer: reworked to split at end, as BQSR should operate on whole bam, not split

migbro committed:
:hammer: removed using interval list for split and bqsr as Broad does not do this

migbro committed:
:wrench: fixed secondary file refrecnes

migbro committed:
:racehorse: added split by read ct for split n trim to improve overall performance

migbro committed:
:wrench: correnct input for apply bqsr

migbro committed:
:wrench: fixed secondary file

migbro committed:
:wrench: several bug fixes

migbro committed:
:racehorse: added aonther split merge step to speed up applybqsr

migbro committed:
:hammer: revert to split on intervals - spltn does use mate
:racehorse: adjust instance type and count

migbro committed:
:twisted_rightwards_arrows: move split n and rename subwf to preprocess
:racehorse: increase num records for split

migbro committed:
:wrench: adding missing refrence fasta to subwf

migbro committed:
:hammer: added default value to scatter_ct, added secondary File to subwf bam output

migbro committed:
:turtle: revert to a more strict interpretation of Broad WDL. Variability introduced with splitting that deviates

migbro committed:
:turtle: turtle: no intervals for split n either

migbro committed:
:hammer: updated outputs - filtered vcf and haplotype called vcf have same content except for FILTER field.
:book: updated readme with ne winputs, outputs, command

migbro committed:
:racehorse: tweaked mem params
:book: updated readme and docs, added antisipated release info

kmhernan commented on subworkflows/preprocess_rnaseq_bam.cwl:
```diff
@@ -1,7 +1,8 @@
 doc: >-
-    "Sambamba Sort Mark Dup Sub WF"
+    "Preprocessing sub workflow for GATK4 RNAseq SNV calling.
+    Uses Sambamba for mark dups and sort, GATK SplitNCigarReads for final processed bam output"
```
While sambamba is much more efficient than Picard, the implementations are not exactly the same. I'm not so concerned about it, but it's important to be aware of this.

migbro replied:
Thanks! I've made more updates in which I use sambamba just once - no intermediate sort steps. As for picard compatibility, I can do some benchmarking, but am also ok with general consensus that the effect is minimal.

migbro replied:
Added a qualifying sentence to the readme. A fair amount of updates from the last PR...I could try and close this one and instead do it by release tag to that PR #1 goes to v0.3, PR # v0.4

migbro committed:
:handshake: Some compaitiblty updates to work with KF bams and NASA bams
:hammer: Use sambamba markdup only
:tada: added VEP100 basic annot, added GATK4 task building

migbro committed:
:book: updated python notebook with task run and useful tagging and adjustement scripts
:hammer: limit sambamba markdup to avoid thread issues
:hammer: added prams to VEP

migbro committed:
:book: updated readme with latest wf diagram and gatk4 wf updates

migbro committed:
Update README.md

kmhernan reviewed (approved):
LGTM 🙏🏽
